### PR TITLE
make it more convenient to create plain/text content with send_as

### PR DESCRIPTION
**Pull request by sdondley**
For #1418, second attempt, to fix earlier botched pull request.

### CODE REVIEW

**Comment by cromedome** [comment]
:+1: Thanks for tackling this!

File: lib/Dancer2/Core/App.pm [hunk 1]
```diff
@@ -942,15 +942,15 @@ sub send_as {
 
     $type or croak "Can not send_as using an undefined type";
 
-    if ( lc($type) eq 'html' ) {
-        if ( $type ne 'html' ) {
+    if ( lc($type) eq 'html' || lc($type) eq 'plain' ) {
+        if ( $type ne 'html' && $type ne 'plain' ) {
```
Comment (yanick) [inline lib/Dancer2/Core/App.pm]:
```suggestion
        if ( $type ne lc $type ) {
```

File: lib/Dancer2/Core/App.pm [hunk 2]
```diff
@@ -942,15 +942,15 @@ sub send_as {
...
             local $Carp::CarpLevel = 2;
-            carp "Please use 'html' as the type for 'send_as', not $type";
+            carp sprintf( "Please use %s as the type for 'send_as', not %s", lc($type), $type );
         }
 
         $options->{charset} = $self->config->{charset} || 'UTF-8';
         my $content = Encode::encode( $options->{charset}, $data );
-        $options->{content_type} ||= 'text/html';
+        $options->{content_type} ||= ($type eq lc('html') ? 'text/html' : 'text/plain');
```
Comment (yanick) [inline lib/Dancer2/Core/App.pm]:
```suggestion
        $options->{content_type} ||= join '/', 'text', lc $type;
```

**Comment by cromedome** [comment]
Merged and released, thanks!

**Pull request closed without merging**